src/include/ipc4/module.h

```diff
@@ -142,7 +142,7 @@ struct ipc4_module_init_instance {
 			uint32_t extended_init      : 1;
 			uint32_t _hw_reserved_2     : 2;
 		} r;
-	} data;
+	} header_ext;
```

Comment:
I think in IPC4 it is more like that we have a `header` which is u64 and it has two u32 parts: `primary` and an `extension`.
The `header` (as two u32) is sent via registers back and forth between the host and DSP.
Additional data is via shared mailbox memory (set/get large).